app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    results_dict = {}
+
+    def inner(*args) -> int:
+        if (*args,) in results_dict:
```

Comment:
The `args` parameter already represents a tuple of arguments, so there is no need to pack it into another tuple `(*args,) `